rockcraft/extensions/springboot.py

```diff
@@ -0,0 +1,226 @@
+            raise ExtensionError(
+                "missing pom.xml and build.gradle file",
+                doc_slug="/reference/extensions/spring-boot-framework",
+                logpath_report=False,
+            )
+        if (self.mvnw_path.exists() and not os.access(self.mvnw_path, os.X_OK)) or (
+            self.gradlew_path.exists() and not os.access(self.gradlew_path, os.X_OK)
+        ):
+            raise ExtensionError(
+                "mvnw or gradlew file is not executable",
+                doc_slug="/reference/extensions/spring-boot-framework",
+                logpath_report=False,
+            )
+
+    @property
+    def mvnw_path(self) -> pathlib.Path:
+        """Return the path to the mvnw file."""
+        return self.project_root / "mvnw"
+
+    @property
+    def pom_xml_path(self) -> pathlib.Path:
+        """Return the path to the pom.xml file."""
+        return self.project_root / "pom.xml"
+
+    @property
+    def gradlew_path(self) -> pathlib.Path:
+        """Return the path to the gradlew file."""
+        return self.project_root / "gradlew"
+
+    @property
+    def build_gradle_path(self) -> pathlib.Path:
+        """Return the path to the build.gradle file."""
+        return self.project_root / "build.gradle"
+
+    @property
+    def init_gradle_path(self) -> pathlib.Path | None:
+        """Return the path to the init.gradle file."""
+        return next(
+            (file for file in self.project_root.rglob("init.gradle") if file.is_file()),
+            None,
+        )
+
+    @property
+    def _rock_base(self) -> str:
+        """Return the base of the rockcraft project."""
+        return self.yaml_data["base"]
+
+    def gen_install_app_part(self) -> dict[str, Any]:
+        """Generate the install-app part."""
+        if self.mvnw_path.exists() and self.gradlew_path.exists():
+            raise ExtensionError(
+                "Both mvnw and gradlew exist. Please remove one of them."
+            )
+
+        user_build_packages_override = (
+            self.yaml_data.get("parts", {})
+            .get("spring-boot-framework/install-app", {})
+            .get("build-packages", [])
+        )
+        plugin: Literal["gradle", "maven"] = (
+            "gradle" if self.build_gradle_path.exists() else "maven"
+        )
+        install_app_part = {
+            "plugin": plugin,
+            "source": ".",
+            "build-packages": (
+                user_build_packages_override
+                if user_build_packages_override
+                else self._gen_install_app_build_packages(plugin=plugin)
+            ),
+            "override-build": "craftctl default\n"
+            "find ${CRAFT_PART_INSTALL} -name '*-plain.jar' -type f -delete",
+            **self._gen_install_app_plugin_directives(plugin=plugin),
+            "organize": {"**/*.jar": "app/"},
+        }
+        if self._rock_base == "bare":
+            install_app_part["stage-packages"] = ["zlib1g", "libstdc++6"]
```

Comment:
for the future: could these be chisel slices?

Reply:
Oh yes definitely, if they are chilseled slices it would also help consolidate them to a single part too!